tests/imapp_benchmark/imapp_bench.go

```diff
@@ -0,0 +1,249 @@
+		cfg.EVMConfig.Debug = false
+
+		b.StartTimer()
+
+		exBytes2, _, err := runtime.Execute(bytecode, calldata, cfg)
+		exBytes = exBytes2
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+	}
+
+	result = exBytes
+}
+
+func BenchmarkBytecodeExecutionIndividual(b *testing.B) {
+	var calldata []byte
+	var exBytes []byte
+
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+
+		cfg := new(runtime.Config)
+		setDefaults(cfg)
+		// from `github.com/ethereum/go-ethereum/core/vm/runtime/runtime.go:109`
+		cfg.State, _ = state.New(common.Hash{}, state.NewDatabase(rawdb.NewMemoryDatabase()), nil)
+
+		// Initialize some constant calldata of 128KB, 2^17 bytes.
+		// This means, if we offset between 0th and 2^16th byte, we can fetch between 0 and 2^16 bytes (64KB)
+		// In consequence, we need args to memory-copying OPCODEs to be between 0 and 2^16, 2^16 fits in a PUSH3,
+		// which we'll be using to generate arguments for those OPCODEs.
+		calldata = []byte(strings.Repeat("{", 1<<17))
+
+		// Warm-up. **NOTE** we're keeping tracing on during warm-up, otherwise measurements are off
+		cfg.EVMConfig.Debug = false
+		_, _, errWarmUp := runtime.Execute(bytecode, calldata, cfg)
+
+		if errWarmUp != nil {
+			b.Error()
+			return
+		}
+
+		b.StartTimer()
+
+		exBytes2, _, err := runtime.Execute(bytecode, calldata, cfg)
+		exBytes = exBytes2
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+	}
+
+	result = exBytes
+}
+
+// // copied directly from github.com/ethereum/go-ethereum/core/vm/runtime/runtime.go
+// // so that we skip this in measured code
+// sets defaults on the config
+func setDefaults(cfg *runtime.Config) {
+	if cfg.ChainConfig == nil {
+		cfg.ChainConfig = &params.ChainConfig{
+			ChainID:             big.NewInt(1),
+			HomesteadBlock:      new(big.Int),
+			DAOForkBlock:        new(big.Int),
+			DAOForkSupport:      false,
+			EIP150Block:         new(big.Int),
+			EIP150Hash:          common.Hash{},
+			EIP155Block:         new(big.Int),
+			EIP158Block:         new(big.Int),
+			ByzantiumBlock:      new(big.Int),
+			ConstantinopleBlock: new(big.Int),
+			PetersburgBlock:     new(big.Int),
+			IstanbulBlock:       new(big.Int),
+			MuirGlacierBlock:    new(big.Int),
+			BerlinBlock:         new(big.Int),
+			LondonBlock:         new(big.Int),
+		}
+	}
+
+	if cfg.Difficulty == nil {
+		cfg.Difficulty = new(big.Int)
+	}
+	if cfg.Time == nil {
+		cfg.Time = big.NewInt(time.Now().Unix())
+	}
+	if cfg.GasLimit == 0 {
+		cfg.GasLimit = math.MaxUint64
+	}
+	if cfg.GasPrice == nil {
+		cfg.GasPrice = new(big.Int)
+	}
+	if cfg.Value == nil {
+		cfg.Value = new(big.Int)
+	}
+	if cfg.BlockNumber == nil {
+		cfg.BlockNumber = new(big.Int)
+	}
+	if cfg.GetHashFn == nil {
+		cfg.GetHashFn = func(n uint64) common.Hash {
+			return common.BytesToHash(crypto.Keccak256([]byte(new(big.Int).SetUint64(n).String())))
+		}
+	}
+	if cfg.BaseFee == nil {
+		cfg.BaseFee = big.NewInt(params.InitialBaseFee)
+	}
+}
+
+func runBenchmark(sampleSize int) {
```

Comment:
if `runBenchmark` or `BenchmarkBytecodeExecutionIndividualWithWarmup` or `BenchmarkBytecodeExecutionIndividual` will not be used by yourself, pls remove to keep things simple. Pick the benchmark method which you think is most reasonable/gives best results. Document the choice

Reply:
Removed and documentation in progress